Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java

```diff
@@ -200,13 +200,22 @@ public void onAnimationFinished() {
 
     @Override
     public void onPaymentMethodNonceCreated(final PaymentMethodNonce paymentMethodNonce) {
-        if (!mRequestedThreeDSecure && paymentMethodNonce instanceof CardNonce &&
+        if (!mPerformedThreeDSecureVerification && paymentMethodNonce instanceof CardNonce &&
```

Comment:
I don't understand this logic, here, and I think it's the naming convention that's throwing me.  If we `performed` 3DS verification, why do they then call `shouldRequest` 3DS verification?

Reply:
`mPerformedThreeDSecureVerification` is a boolean that is set to tru when a card is 3DS verified in order to keep from reverifying the same card after it's just been verified.  If for some reason the 3DS Flow does not complete (`onError` or `onCancel` are called), `handleThreeDSecureFailure()` is called and will set it back to false so the next attempt will again go through 3DS.

`shouldRequestThreeDSecureVerification()` on the other hand checks that `dropInRequest.requestThreeDSecureVerification()` was set to true, that `dropInRequest.amount()` is set to a value, and finally that the merchant is configured for 3DS.